domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1108 @@
+   *   to the default domain, example.org.
+   *
+   * @usage drush domain-delete --chatty all
+   *   Verbosely Delete the domains example.com and example.net and assign
+   *   their content and users to the default domain, example.org.
+   *
+   * @option chatty
+   *   Document each step as it is performed.
+   * @option dryrun
+   *   Do not do anything, but explain what would be done. Implies --chatty.
+   * @option content-as
+   *   Values "prompt", "ignore", "default", <name>, Reassign content
+   *   associated with the the domain being deleted to the default domain, to
+   *   a specified domain, or leave the content alone (&hence inaccessible in
+   *   the normal way). The default value is 'prompt': ask which domain to use.
+   * @option users-as
+   *   Values "prompt", "ignore", "default", <name>, Reassign user accounts
+   *   associated with the the domain being deleted to the default domain,
+   *   to the domain whose machine name is <name>, or leave the user accounts
+   *   alone (&hence inaccessible in the normal way). The default value is
+   *   'prompt': ask which domain to use.
+   *
+   * @command domain:delete
+   * @aliases domain-delete
+   * 
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function delete($domain_id, $options = ['content-as' => null, 'users-as' => null, 'dryrun' => null, 'chatty' => null ]) {
+    $policy_content = 'prompt';
+    $policy_users = 'prompt';
+
+    //region Get current domain list and perform sanity checks.
+    $domain_storage = $this->getStorage();
+    $default_domain = $domain_storage->loadDefaultDomain();
+    $all_domains = $domain_storage->loadMultipleSorted(NULL);
+
+    if (empty($all_domains)) {
+      throw new DomainCommandException('There are no configured domains.');
+    }
+    if (is_null($domain_id)) {
+      throw new DomainCommandException('You must specify a domain to delete.');
+    }
+    //endregion
+
+    //region Determine which domains to be deleted.
+    if ($domain_id == 'all') {
+      $domains = $all_domains;
+      if (empty($domains)) {
+        $this->logger()->info(dt('There are no domains to delete.'));
+        return;
+      }
+      $really = $this->io()->confirm(dt('This action cannot be undone. Continue?:'), FALSE);
+      if (empty($really)) {
+        return;
+      }
+    }
+    elseif ($domain = $this->getDomainFromArgument($domain_id)) {
+      if ($domain->isDefault()) {
+        throw new DomainCommandException('The primary domain may not be deleted. '
+                                         .'Use drush domain:default to set a new default domain.');
+      }
+      $domains = [$domain];
+    }
+    else {
+      $this->logger()->info(dt('Nothing to do?'));
+      return;
+    }
+    // Create list of domain (entity) ids to delete.
+    $delete_domain_ids = array_map(function($v){return $v->id;}, $domains);
+    //endregion
+
+    //region Get content disposition from configuration and validate.
```

Comment:
These regions likely need to be calls to the registry we discussed, where each field acts in it's own method. Those methods like go in `DomainElementManager` and are modeled on `getFieldValues`

Reply:
I'll have a think about this too.